pkg/controllers/provisioning/provisioner.go

```diff
@@ -203,12 +207,15 @@ func (p *Provisioner) launch(ctx context.Context, node *scheduling.Node) error {
 	if err := latest.Spec.Limits.ExceededBy(latest.Status.Resources); err != nil {
 		return err
 	}
+
+	// apply both the taints and startup taints to the node
+	taints := append(node.Provisioner.Spec.Taints, node.Provisioner.Spec.StartupTaints...)
```

Comment:
nit: inline?

Reply:
I could go either way, but I think it's slightly less clear inlined.